src/wp-includes/media.php

```diff
@@ -1785,6 +1784,36 @@ function wp_lazy_loading_enabled( $tag_name, $context ) {
 	return (bool) apply_filters( 'wp_lazy_loading_enabled', $default, $tag_name, $context );
 }
 
+/**
+ * Determines whether to add the `decoding` attribute to the specified tag in the specified context.
+ *
+ * @since 6.4.0
+ *
+ * @param string $tag_name The tag name.
+ * @param string $context  Additional context, like the current filter name
+ *                         or the function name from where this was called.
+ * @return bool Whether to add the attribute.
+ */
+function wp_decoding_enabled( $tag_name, $context ) {
```

Comment:
Is a standalone function for this necessary, or can this logic live in `wp_get_loading_optimization_attributes` where it is used? If it's a requirement so we aren't repeating logic in more than one place, or some other reason, then we should consider a more specific name, like `wp_img_decoding_attr_enabled()`. I also wonder if we need the `$tag_name` param here, since I believe this is only applicable to images.

Reply:
@joemcgill 

The intent here was to mimic — if nothing else for consistency — `wp_lazy_loading_enabled()` which deals with a different attribute in the optimization scope. But I agree that this is unnecessary. I'll take it out.